src/openms/source/ANALYSIS/OPENSWATH/OpenSwathScoring.cpp

```diff
@@ -252,27 +282,35 @@ namespace OpenMS
       im_range.clear();
     }
 
-    // Identify corresponding SONAR maps (if more than one map is used)
-    std::vector<OpenSwath::SwathMap> used_swath_maps;
-    if (swath_maps.size() > 1)
+    // If add_up_spectra_ is -1, we automatically compute the amount of spectra to add by setting add_up_spectra_ to the ceiling value of N percent of the retention time peak width
+    if (add_up_spectra_ == -1)
     {
-      double precursor_mz = transition.getPrecursorMZ();
-      for (size_t i = 0; i < swath_maps.size(); ++i)
-      {
-        if (swath_maps[i].ms1) {continue;} // skip MS1
-        if (precursor_mz > swath_maps[i].lower && precursor_mz < swath_maps[i].upper)
-        {
-          used_swath_maps.push_back(swath_maps[i]);
+        double leftWidth = imrmfeature->getMetaValue("leftWidth");
+        double rightWidth = imrmfeature->getMetaValue("rightWidth");
+        double peakWidth = rightWidth - leftWidth;
+        add_up_spectra_ = std::ceil(peakWidth * use_percent_peak_width_);
+        
+        // Ensure add_up_spectra_ is odd
+        if (static_cast<int>(add_up_spectra_) % 2 == 0) {
+            add_up_spectra_ -= 1;
         }
-      }
-    }
-    else
-    {
-      used_swath_maps = swath_maps;
+        
+        // Ensure add_up_spectra_ is at least 1 if it was set to <= 0 based on the peak width
+        if (add_up_spectra_ <= 0) {
+            add_up_spectra_ = 1;
+        }
+        
+        OPENMS_LOG_DEBUG 
+            << "Fetching Spectra between leftWidth: " << leftWidth 
+            << " rightWidth: " << rightWidth 
+            << " peakWidth: " << peakWidth 
+            << " add_up_spectra_: " << add_up_spectra_ 
+            << " use_percent_peak_width_: " << use_percent_peak_width_ 
+            << std::endl;
     }
 
     // find spectrum that is closest to the apex of the peak using binary search
-    std::vector<OpenSwath::SpectrumPtr> spectrum = fetchSpectrumSwath(used_swath_maps, imrmfeature->getRT(), add_up_spectra_, im_range);
+    std::vector<OpenSwath::SpectrumPtr> spectrum = fetchSpectrumSwath(swath_maps, imrmfeature->getRT(), add_up_spectra_, im_range);
```

Comment:
would be better to pass a local variable instead of a modified member.
the member variable should represent a property of the scoring object but add_up_spectra is now a property that depends on the individual feature.

Reply:
so maybe a local variable spectra_merge_window that gets set based on the add_up_spectra_ / percentage settings